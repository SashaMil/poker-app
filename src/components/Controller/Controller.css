.controllerH3{
  color: white;
}
