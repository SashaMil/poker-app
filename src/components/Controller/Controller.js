import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import {Link} from 'react-router-dom';
import Slide from '@material-ui/core/Slide';
import Button from '@material-ui/core/Button';
import { computerDecision } from '../../redux/actions/tableActions';
import Slider from '@material-ui/lab/Slider';
import Typography from '@material-ui/core/Typography';


import './Controller.css';

const mapStateToProps = state => ({
  user: state.user,
  table: state.table,
});

class Controller extends Component {

  componentDidMount = () => {
    console.log(this.props.currentAction.bet);
  }
  componentDidUpdate = () => {
    console.log(this.props.currentAction.bet);
  }

  render() {
    return (
      <div>
        {this.props.currentAction.bet > 0 ? (
          <div>
            <div>
              <Button onClick={this.props.fold} variant="contained" color="secondary">
                Fold
              </Button>
            </div>
            <div>
              <Button onClick={this.props.call} variant="contained" color="default">
                Call
              </Button>
            </div>
            <div>
              <Button onClick={() => this.props.raise(this.props.value)} variant="contained" color="primary">
                Raise
              </Button>
            </div>
          </div>
        ) : (
          <div>
            <div>
              <Button onClick={this.props.check} variant="contained" color="secondary">
                Check
              </Button>
            </div>
            <div>
              <Button onClick={() => this.props.bet(this.props.value)} variant="contained" color="secondary">
                Bet
              </Button>
            </div>
          </div>
        )
      }
        <div>
          <Slider value={this.props.value} min={10} max={2000} step={5} onChange={(event, value) => this.props.handleChange(event,value)} />
        </div>
        <Typography variant="headline" style={{ color: 'white' }}>Current Player Bet: {this.props.value}</Typography>

      </div>
    )
  }
}

Slide.defaultProps = {
  timeout: {
    enter: 1000,
    exit: 1000,
  },
};


export default compose(connect(mapStateToProps))(Controller);
