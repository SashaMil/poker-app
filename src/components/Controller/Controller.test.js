import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Controller from './Controller.js';

const mockStore = configureMockStore();
const store = mockStore({});

describe("Controller", () => {
  it("should check state, which is equivalent to passed props", () => {
        expect(
            shallow(
                <Provider store={store}>
                    <Controller />
                </Provider>
            )
        );
    });
});
