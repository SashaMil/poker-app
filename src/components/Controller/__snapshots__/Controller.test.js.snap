// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller should check state, which is equivalent to passed props 1`] = `<Connect(Controller) />`;

exports[`Controller should render without throwing an error 1`] = `<Connect(Controller) />`;
